// npmvc-express-example-route/src/view/RouteMediator.js
var fs = require('fs');
var path = require('path');

module.exports = function(include, puremvc) {

	puremvc.include("IndexRoute", __dirname + "/route");
	puremvc.include("SomeRoute", __dirname + "/route");

	var ExpressMediator = npmvc.view.mediator.ExpressMediator;

	/**
	 * @class npmvc.view.mediator.IndexRouteMediator
	 * @extends puremvc.Mediator
	 * @imports
	 */
	puremvc.define(
		// CLASS INFO
		{
			name: 'npmvc.view.mediator.RouteMediator',
			parent: puremvc.Mediator
		},

		// INSTANCE MEMBERS
		{
			/** @override */
			onRegister: function() {
				this.expressMediator = this.facade.retrieveMediator(npmvc.view.mediator.ExpressMediator.NAME);

				// Use handlebars component
				this.expressMediator.addComponent("npmvc-express-middleware-handlebars", {
					template_folder: __dirname + "/../../",
					cache: false
				});

				// Use handlebars component
				this.expressMediator.addComponent("npmvc-express-middleware-static-files", {
					path: __dirname + "/../../static"
				});
			},
			/** @override */
			listNotificationInterests: function() {
				return [npmvc.view.mediator.ExpressMediator.EXPRESS_SETUP_ROUTES, npmvc.view.mediator.ExpressMediator.EXPRESS_READY]
			},
			/** @override */
			handleNotification: function(note) {
				switch (note.getName()) {
					case npmvc.view.mediator.ExpressMediator.EXPRESS_SETUP_ROUTES:

						// initiate and register routes here...
						var indexRoute = new my.custom.IndexRoute();
						this.facade.registerProxy(indexRoute);

						var someRoute = new my.custom.SomeRoute();
						this.facade.registerProxy(someRoute);

						break;
					case npmvc.view.mediator.ExpressMediator.EXPRESS_READY:
						
					break;
				}
			},
			/**
			 * [init description]
			 * @return {[type]} [description]
			 */
			init: function(config) {
				this.config = config;
			}
		},

		// STATIC MEMBERS
		{
			/**
			 * @static
			 * @type {string}
			 */
			NAME: 'npmvc.view.mediator.RouteMediator',
		});
}
